interfaces/inc/c_pal/async_socket.h

```diff
@@ -76,6 +73,10 @@ typedef struct ASYNC_SOCKET_BUFFER_TAG
     uint32_t length;
 } ASYNC_SOCKET_BUFFER;
 
+#include "umock_c/umock_c_prod.h"
```

Comment:
Consider having the include up there next to the other ones. Having includes as close to the top as possible can reduce developer annoyance  - because who scrolls down for includes :-). /consider

Reply:
I considered, and this is our very special umock_c_prod which should ideally encompass only APIs (which is what this does here). I think I like it to stay here rather than someplace where it can mangle with other headers.